public/coffee/ide/editor/controllers/SavingNotificationController.coffee

```diff
@@ -13,21 +13,25 @@ define [
 		$scope.docSavingStatus = {}
 		pollSavedStatus = () ->
 			oldStatus = $scope.docSavingStatus
-			$scope.docSavingStatus = {}
+			newStatus = {}
 
 			for doc_id, doc of Document.openDocs
 				saving = doc.pollSavedStatus()
 				if !saving
 					if oldStatus[doc_id]?
-						$scope.docSavingStatus[doc_id] = oldStatus[doc_id]
-						$scope.docSavingStatus[doc_id].unsavedSeconds += 1
+						newStatus[doc_id] = oldStatus[doc_id]
+						newStatus[doc_id].unsavedSeconds += 1
 					else
-						$scope.docSavingStatus[doc_id] = {
+						newStatus[doc_id] = {
 							unsavedSeconds: 0
 							doc: ide.fileTreeManager.findEntityById(doc_id)
 						}
 
+			if _.size(newStatus) or _.size(oldStatus)
```

Comment:
I don't understand what this is testing. Can you make it explicit with a comment or refactor?